contrib/dracut/90zfs/module-setup.sh.in

```diff
@@ -105,28 +105,48 @@ install() {
 	if dracut_module_included "systemd"; then
 		mkdir -p "${initdir}/$systemdsystemunitdir/zfs-import.target.wants"
 		for _item in scan cache ; do
-			dracut_install @systemdunitdir@/zfs-import-$_item.service
-			if ! [ -L "${initdir}/$systemdsystemunitdir/zfs-import.target.wants"/zfs-import-$_item.service ]; then
-				ln -s ../zfs-import-$_item.service "${initdir}/$systemdsystemunitdir/zfs-import.target.wants"/zfs-import-$_item.service
-				type mark_hostonly >/dev/null 2>&1 && mark_hostonly @systemdunitdir@/zfs-import-$_item.service
+			_service="zfs-import-$_item.service"
+			dracut_install "@systemdunitdir@/$_service"
+
+			mkdir -p "${initdir}/$systemdsystemunitdir/$_service.d"
+			{
+				echo "[Service]"
+				echo "ExecStart="
+				awk '/^ExecStart=/ {print $0 " $ZPOOL_IMPORT_OPTS"}' "@systemdunitdir@/$_service"
+			} > "${initdir}/$systemdsystemunitdir/$_service.d/import-opts.conf"
```

Comment:
What does this block do since we don't seem to consume the "import-opts.conf"? I could be missing something.

Also `echo "ExecStart="` is extraneous since subsequent `awk` command prints the entire "ExecStart" line.

Neat investigative work!

Reply:
You are, on both counts: this is a drop-in for the systemd service. 

That being said, I no longer think this is the right approach (since it's (a) hacky and (b) kinda magic), and I've instead opted for adding `${ZPOOL_IMPORT_OPTS}` to both services themselves (previous HEAD at [suseł-but-stronger-v2-v1](https://github.com/nabijaczleweli/zfs/tree/suse%C5%82-but-stronger-v2-v1)).

Reply:
Nice. I agree the new approach is cleaner. Thanks!